library/src/main/java/net/jimblackler/jsonschemafriend/FormatChecker.java

```diff
@@ -48,13 +49,13 @@
   }
 
   static String formatCheck(
-      String string, String format, URI metaSchema, RegExPatternSupplier regExPatternSupplier) {
+      String string, String format, URI metaSchema, RegExPatternSupplier regExPatternSupplier, boolean validateFormats) {
     boolean preDraft4 = metaSchema.equals(DRAFT_3);
     boolean preDraft6 = preDraft4 || metaSchema.equals(DRAFT_4);
     boolean preDraft7 = preDraft6 || metaSchema.equals(DRAFT_6);
     boolean preDraft2019 = preDraft7 || metaSchema.equals(DRAFT_7);
 
-    if (!preDraft7 && preDraft2019) {
+    if ((!preDraft7 && preDraft2019) || validateFormats) {
```

Comment:
Do we care about the combination of `preDraft7 == true` and `validateFormats == true`?